modules/oe_content_redirect_link_field/tests/src/Functional/RedirectSubscriberTest.php

```diff
@@ -0,0 +1,250 @@
+
+    // Add a translation and assert the redirect to the correct path.
+    $translated_node = $external_node->addTranslation('fr', $external_node->toArray());
+    $translated_node->set('oe_redirect_link', 'http://example.com/french');
+    $translated_node->save();
+    $this->assertRedirect('http://example.com/french', $external_node, 'MISS', 301, 'fr');
+    $this->assertRedirect('http://example.com/3', $external_node, 'MISS');
+
+    // Make another request to check the response was cached.
+    $this->assertRedirect('http://example.com/french', $external_node, 'HIT', 301, 'fr');
+    $this->assertRedirect('http://example.com/3', $external_node, 'HIT');
+
+    // Unset the redirect link from the source translation.
+    $external_node->set('oe_redirect_link', NULL);
+    $external_node->save();
+    $this->drupalGet('/node/' . $external_node->id());
+    $this->assertSession()->addressEquals('/node/' . $external_node->id());
+    $this->drupalGet('/fr/node/' . $translated_node->id());
+    $this->assertSession()->addressEquals('/fr/node/' . $translated_node->id());
+
+    // Create a node that redirects to another node whose alias changes and
+    // assert the redirect is cached correctly.
+    $target_node = $this->drupalCreateNode([
+      'type' => 'test_with_redirect_link',
+      'path' => ['alias' => '/target-node'],
+      'status' => NodeInterface::PUBLISHED,
+    ]);
+
+    $target_node->save();
+
+    $redirect_node = $this->drupalCreateNode([
+      'type' => 'test_with_redirect_link',
+      'oe_redirect_link' => 'entity:node/' . $target_node->id(),
+      'status' => NodeInterface::PUBLISHED,
+    ]);
+    $redirect_node->save();
+    $this->assertRedirect(Url::fromUserInput('/target-node')->setAbsolute()->toString(), $redirect_node, 'MISS');
+    $response = $this->assertRedirect(Url::fromUserInput('/target-node')->setAbsolute()->toString(), $redirect_node, 'HIT');
+    $this->assertContains('node:' . $target_node->id(), explode(' ', $response->getHeaders()['X-Drupal-Cache-Tags'][0]));
+
+    $target_node->set('path', ['alias' => '/new-path']);
+    $target_node->save();
+    $this->assertRedirect(Url::fromUserInput('/new-path')->setAbsolute()->toString(), $redirect_node, 'MISS');
+  }
+
+  /**
+   * Asserts the redirect of a node to a certain path.
+   *
+   * @param string $expected
+   *   The expected redirect path.
+   * @param \Drupal\node\NodeInterface $node
+   *   The node being tested for redirect.
+   * @param string $cache
+   *   Whether the Drupal cache was HIT or MISS-ed.
+   * @param int|null $expected_status_code
+   *   The expected status code.
+   * @param string|null $language
+   *   The expected path language.
+   *
+   * @return \Psr\Http\Message\ResponseInterface
+   *   The response.
+   */
+  public function assertRedirect(string $expected, NodeInterface $node, string $cache, int $expected_status_code = 301, string $language = NULL): ResponseInterface {
```

Comment:
The $cache parameter is not used, so the cache hit checks are missing.

Reply:
Man, need a holiday. Fixed.